lte/gateway/python/integ_tests/s1aptests/test_no_esm_information_rsp_with_mme_restart.py

```diff
@@ -0,0 +1,144 @@
+        eps_type = s1ap_types.TFW_EPS_ATTACH_TYPE_EPS_ATTACH
+        attach_req.mIdType = id_type
+        attach_req.epsAttachType = eps_type
+        attach_req.useOldSecCtxt = sec_ctxt
+
+        # enabling ESM Information transfer flag
+        attach_req.eti.pres = 1
+        attach_req.eti.esm_info_transfer_flag = 1
+
+        print("Sending Attach Request ue-id", ue_id)
+        self._s1ap_wrapper._s1_util.issue_cmd(
+            s1ap_types.tfwCmd.UE_ATTACH_REQUEST, attach_req
+        )
+
+        response = self._s1ap_wrapper.s1_util.get_response()
+        self.assertEqual(
+            response.msg_type, s1ap_types.tfwCmd.UE_AUTH_REQ_IND.value
+        )
+        print("Received Auth Req for ue-id", ue_id)
+
+        auth_res = s1ap_types.ueAuthResp_t()
+        auth_res.ue_Id = ue_id
+        sqn_recvd = s1ap_types.ueSqnRcvd_t()
+        sqn_recvd.pres = 0
+        auth_res.sqnRcvd = sqn_recvd
+        print("Sending Auth Response for ue-id", ue_id)
+        self._s1ap_wrapper._s1_util.issue_cmd(
+            s1ap_types.tfwCmd.UE_AUTH_RESP, auth_res
+        )
+
+        response = self._s1ap_wrapper.s1_util.get_response()
+        self.assertEqual(
+            response.msg_type, s1ap_types.tfwCmd.UE_SEC_MOD_CMD_IND.value
+        )
+        print("Received Security Mode Command for ue-id", ue_id)
+
+        sec_mode_complete = s1ap_types.ueSecModeComplete_t()
+        sec_mode_complete.ue_Id = ue_id
+        self._s1ap_wrapper._s1_util.issue_cmd(
+            s1ap_types.tfwCmd.UE_SEC_MOD_COMPLETE, sec_mode_complete
+        )
+        print("Sent Security Mode Complete for ue-id", ue_id)
+
+        # Esm Information Request indication
+        print("Received Esm Information Request ue-id", ue_id)
+        response = self._s1ap_wrapper.s1_util.get_response()
+        self.assertEqual(
+            response.msg_type, s1ap_types.tfwCmd.UE_ESM_INFORMATION_REQ.value
+        )
+
+        print("************************* Restarting MME service on", "gateway")
+        self._s1ap_wrapper.magmad_util.restart_services(["mme"])
+
+        for j in range(30):
+            print("Waiting for", j, "seconds")
+            time.sleep(1)
+
+        # Receive UE_ESM_INFORMATION_REQ, as sometimes MME retransmits
+        # UE_ESM_INFORMATION_REQ message before it restarts
+        # as T3489 timer is only 4 secs.
+        response = self._s1ap_wrapper.s1_util.get_response()
+        if response.msg_type == s1ap_types.tfwCmd.UE_ESM_INFORMATION_REQ.value:
+            print(
+                "Received retransmitted UE_ESM_INFORMATION_REQ for ue-id",
+                ue_id,
+            )
+            # Wait for UE_CTX_REL_IND
+            response = self._s1ap_wrapper.s1_util.get_response()
+            self.assertEqual(
+                response.msg_type, s1ap_types.tfwCmd.UE_CTX_REL_IND.value
+            )
+        else:
+            self.assertEqual(
+                response.msg_type, s1ap_types.tfwCmd.UE_CTX_REL_IND.value
+            )
```

Comment:
We can skip the else condition here and just have one assert
```suggestion
        self.assertEqual(
            response.msg_type, s1ap_types.tfwCmd.UE_CTX_REL_IND.value
        )
       
```